### PR TITLE
fix(tsdb): attempt to work on docker on windows

### LINKED ISSUE #9630: thousands of .tsm files created with "invalid argument" error, disk filled
### Bug report

__System info:__ influxdb on docker on windows 10 64bit, NFLUXDB_VERSION	1.5.1, image  influxdb:latest: sha256:f4d8a0378c2dd057c0867fbd84d67e584f8274cbfd47f851467e6c23a2b0be40, started with -v /g/dati:/var/lib/influxdb 

__Steps to reproduce:__

1. docker run fires up the server
2. create a shell, start influx cli, create db mydb
3. start sending data with a node app:
```
sendPoints(points){
 const influx = new Influx.InfluxDB({
            host: config.influxHost,
            database: config.influxDataBase
          });

           influx.writePoints(points)
                .then(()=>{
             
            },(err)=>{log.log(err+" writing to influx")});
        }   
```
__Expected behavior:__ 

normal working

__Actual behavior:__ 
g:\dati\data\mydb\autogen\2 gets filled with thousands of 000000xxxx-000000001.tsm files of about 2730kb each. even if no data is sent anymore to the server, the tsm files keep being created until full exhaustion of the disk space.

__Additional info:__ 
this logs are sync with the file creation.
if I stop the container, ifluxdb shuts down cleanly.

```
ts=2018-03-24T09:39:52.189289Z lvl=info msg="Cache snapshot (start)" log_id=071qT02G000 engine=tsm1 trace_id=071quE~G000 op_name=tsm1_cache_snapshot op_event=start
ts=2018-03-24T09:39:52.298005Z lvl=info msg="Error adding new TSM files from snapshot" log_id=071qT02G000 engine=tsm1 trace_id=071quE~G000 op_name=tsm1_cache_snapshot error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:52.298044Z lvl=info msg="Snapshot for path written" log_id=071qT02G000 engine=tsm1 trace_id=071quE~G000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/mydb/autogen/2 duration=108.773ms
ts=2018-03-24T09:39:52.298057Z lvl=info msg="Cache snapshot (end)" log_id=071qT02G000 engine=tsm1 trace_id=071quE~G000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=108.783ms
ts=2018-03-24T09:39:52.298062Z lvl=info msg="Error writing snapshot" log_id=071qT02G000 engine=tsm1 error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:53.189252Z lvl=info msg="Cache snapshot (start)" log_id=071qT02G000 engine=tsm1 trace_id=071quIuG000 op_name=tsm1_cache_snapshot op_event=start
ts=2018-03-24T09:39:53.288073Z lvl=info msg="Error adding new TSM files from snapshot" log_id=071qT02G000 engine=tsm1 trace_id=071quIuG000 op_name=tsm1_cache_snapshot error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:53.288109Z lvl=info msg="Snapshot for path written" log_id=071qT02G000 engine=tsm1 trace_id=071quIuG000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/mydb/autogen/2 duration=98.874ms
ts=2018-03-24T09:39:53.288121Z lvl=info msg="Cache snapshot (end)" log_id=071qT02G000 engine=tsm1 trace_id=071quIuG000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=98.879ms
ts=2018-03-24T09:39:53.288126Z lvl=info msg="Error writing snapshot" log_id=071qT02G000 engine=tsm1 error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:54.189247Z lvl=info msg="Cache snapshot (start)" log_id=071qT02G000 engine=tsm1 trace_id=071quMoG000 op_name=tsm1_cache_snapshot op_event=start
ts=2018-03-24T09:39:54.276133Z lvl=info msg="Error adding new TSM files from snapshot" log_id=071qT02G000 engine=tsm1 trace_id=071quMoG000 op_name=tsm1_cache_snapshot error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:54.276169Z lvl=info msg="Snapshot for path written" log_id=071qT02G000 engine=tsm1 trace_id=071quMoG000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/mydb/autogen/2 duration=86.931ms
ts=2018-03-24T09:39:54.276181Z lvl=info msg="Cache snapshot (end)" log_id=071qT02G000 engine=tsm1 trace_id=071quMoG000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=86.940ms
ts=2018-03-24T09:39:54.276189Z lvl=info msg="Error writing snapshot" log_id=071qT02G000 engine=tsm1 error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
[httpd] 172.17.0.1 - root [24/Mar/2018:09:39:55 +0000] "POST /write?db=mydb&p=%5BREDACTED%5D&precision=n&rp=&u=root HTTP/1.1" 204 0 "-" "-" 4f136889-2f47-11e8-806f-000000000000 72407
ts=2018-03-24T09:39:55.189163Z lvl=info msg="Cache snapshot (start)" log_id=071qT02G000 engine=tsm1 trace_id=071quQiG000 op_name=tsm1_cache_snapshot op_event=start
ts=2018-03-24T09:39:55.301790Z lvl=info msg="Error adding new TSM files from snapshot" log_id=071qT02G000 engine=tsm1 trace_id=071quQiG000 op_name=tsm1_cache_snapshot error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:55.301937Z lvl=info msg="Snapshot for path written" log_id=071qT02G000 engine=tsm1 trace_id=071quQiG000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/mydb/autogen/2 duration=112.784ms
ts=2018-03-24T09:39:55.301953Z lvl=info msg="Cache snapshot (end)" log_id=071qT02G000 engine=tsm1 trace_id=071quQiG000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=112.797ms
ts=2018-03-24T09:39:55.301958Z lvl=info msg="Error writing snapshot" log_id=071qT02G000 engine=tsm1 error="sync /var/lib/influxdb/data/mydb/autogen/2: invalid argument"
ts=2018-03-24T09:39:56.189246Z lvl=info msg="Cache snapshot (start)" log_id=071qT02G000 engine=tsm1 trace_id=071quUcG000 op_name=tsm1_cache_snapshot op_event=start
```

### LINKED ISSUE #9833: Does not appear to work in a Linux Docker container on Windows
### Bug report

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]
ts=2018-05-11T19:13:53.903815Z lvl=info msg="InfluxDB starting" log_id=0809pOwl000 version=1.5.2 branch=1.5 commit=02d7d4f043b34ecb4e9b2dbec298c6f9450c2a32
ts=2018-05-11T19:13:53.903975Z lvl=info msg="Go runtime" log_id=0809pOwl000 version=go1.9.2 maxprocs=2

Windows 10 Pro 1709 (build 16299.431)
Docker 18.03.1-ce

__Steps to reproduce:__

1. Using Docker-Compose to create a container for influxdb and another for grafana
2. Using Telgraf for Windows to write to influxdb
3. Works fine from a clean state
4. Restart computer
5. Wait for Docker to be ready
6. docker-compose up

__Expected behavior:
grafana to show the data from earlier when the system was up with a gap for the time it was down

__Actual behavior:
while the logs indicate influxdb shut down gracefully, it was unable to start correctly.

__Additional info:__ [Include gist of relevant config, logs, etc.]
[httpd] 172.21.0.1 - - [11/May/2018:19:09:53 +0000] "POST /write?db=telegraf HTTP/1.1" 204 0 "-" "telegraf" e28dffb0-554e-11e8-84ba-000000000000 24829
[httpd] 172.21.0.1 - - [11/May/2018:19:10:03 +0000] "POST /write?db=telegraf HTTP/1.1" 204 0 "-" "telegraf" e883a87b-554e-11e8-84bb-000000000000 26254
[httpd] 172.21.0.1 - - [11/May/2018:19:10:13 +0000] "POST /write?db=telegraf HTTP/1.1" 204 0 "-" "telegraf" ee79a8a4-554e-11e8-84bc-000000000000 25816
ts=2018-05-11T19:11:16.518646Z lvl=info msg="Signal received, initializing clean shutdown..." log_id=07~z8gGl000
ts=2018-05-11T19:11:16.518681Z lvl=info msg="Waiting for clean shutdown..." log_id=07~z8gGl000
ts=2018-05-11T19:11:16.518701Z lvl=info msg="Shutting down monitor service" log_id=07~z8gGl000 service=monitor
ts=2018-05-11T19:11:16.518753Z lvl=info msg="Listener closed" log_id=07~z8gGl000 service=snapshot
ts=2018-05-11T19:11:16.518760Z lvl=info msg="Terminating storage of statistics" log_id=07~z8gGl000 service=monitor
ts=2018-05-11T19:11:16.518769Z lvl=info msg="Terminating precreation service" log_id=07~z8gGl000 service=shard-precreation
ts=2018-05-11T19:11:16.518777Z lvl=info msg="Terminating continuous query service" log_id=07~z8gGl000 service=continuous_querier
ts=2018-05-11T19:11:16.518789Z lvl=info msg="Closing retention policy enforcement service" log_id=07~z8gGl000 service=retention
ts=2018-05-11T19:13:53.903815Z lvl=info msg="InfluxDB starting" log_id=0809pOwl000 version=1.5.2 branch=1.5 commit=02d7d4f043b34ecb4e9b2dbec298c6f9450c2a32
ts=2018-05-11T19:13:53.903975Z lvl=info msg="Go runtime" log_id=0809pOwl000 version=go1.9.2 maxprocs=2
ts=2018-05-11T19:13:54.108029Z lvl=info msg="Using data dir" log_id=0809pOwl000 service=store path=/var/lib/influxdb/data
ts=2018-05-11T19:13:54.109450Z lvl=info msg="Open store (start)" log_id=0809pOwl000 service=store trace_id=0809pPkG000 op_name=tsdb_open op_event=start
ts=2018-05-11T19:13:54.303542Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:13:54.816503Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/telegraf/autogen/2/_00001.wal size=2391011
ts=2018-05-11T19:13:55.114607Z lvl=info msg="Failed to open shard" log_id=0809pOwl000 service=store trace_id=0809pPkG000 op_name=tsdb_open db_shard_id=2 error="[shard 2] sync /var/lib/influxdb/data/telegraf/autogen/2: invalid argument"
ts=2018-05-11T19:13:55.124054Z lvl=info msg="Failed to open shard" log_id=0809pOwl000 service=store trace_id=0809pPkG000 op_name=tsdb_open db_shard_id=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:13:55.124168Z lvl=info msg="Open store (end)" log_id=0809pOwl000 service=store trace_id=0809pPkG000 op_name=tsdb_open op_event=end op_elapsed=1014.784ms
ts=2018-05-11T19:13:55.124202Z lvl=info msg="Opened service" log_id=0809pOwl000 service=subscriber
ts=2018-05-11T19:13:55.124263Z lvl=info msg="Starting monitor service" log_id=0809pOwl000 service=monitor
ts=2018-05-11T19:13:55.124283Z lvl=info msg="Registered diagnostics client" log_id=0809pOwl000 service=monitor name=build
ts=2018-05-11T19:13:55.124289Z lvl=info msg="Registered diagnostics client" log_id=0809pOwl000 service=monitor name=runtime
ts=2018-05-11T19:13:55.124294Z lvl=info msg="Registered diagnostics client" log_id=0809pOwl000 service=monitor name=network
ts=2018-05-11T19:13:55.124501Z lvl=info msg="Registered diagnostics client" log_id=0809pOwl000 service=monitor name=system
ts=2018-05-11T19:13:55.124718Z lvl=info msg="Starting precreation service" log_id=0809pOwl000 service=shard-precreation check_interval=10m advance_period=30m
ts=2018-05-11T19:13:55.124738Z lvl=info msg="Starting snapshot service" log_id=0809pOwl000 service=snapshot
ts=2018-05-11T19:13:55.124743Z lvl=info msg="Starting continuous query service" log_id=0809pOwl000 service=continuous_querier
ts=2018-05-11T19:13:55.124902Z lvl=info msg="Starting HTTP service" log_id=0809pOwl000 service=httpd authentication=false
ts=2018-05-11T19:13:55.124911Z lvl=info msg="opened HTTP access log" log_id=0809pOwl000 service=httpd path=stderr
ts=2018-05-11T19:13:55.125379Z lvl=info msg="Listening on HTTP" log_id=0809pOwl000 service=httpd addr=[::]:8086 https=false
ts=2018-05-11T19:13:55.125420Z lvl=info msg="Starting retention policy enforcement service" log_id=0809pOwl000 service=retention check_interval=30m
ts=2018-05-11T19:13:55.124893Z lvl=info msg="Storing statistics" log_id=0809pOwl000 service=monitor db_instance=_internal db_rp=monitor interval=10s
ts=2018-05-11T19:13:55.133319Z lvl=info msg="Listening for signals" log_id=0809pOwl000
ts=2018-05-11T19:13:55.136564Z lvl=info msg="Sending usage statistics to usage.influxdata.com" log_id=0809pOwl000
ts=2018-05-11T19:14:10.024165Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:14:10.657776Z lvl=info msg="Write failed" log_id=0809pOwl000 service=write shard=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:10.658092Z lvl=info msg="failed to store statistics" log_id=0809pOwl000 service=monitor error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:20.051376Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:14:20.306750Z lvl=info msg="Write failed" log_id=0809pOwl000 service=write shard=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:20.306961Z lvl=info msg="failed to store statistics" log_id=0809pOwl000 service=monitor error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:30.038802Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:14:30.308278Z lvl=info msg="Write failed" log_id=0809pOwl000 service=write shard=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:30.308347Z lvl=info msg="failed to store statistics" log_id=0809pOwl000 service=monitor error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:40.042630Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:14:40.295905Z lvl=info msg="Write failed" log_id=0809pOwl000 service=write shard=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:40.295961Z lvl=info msg="failed to store statistics" log_id=0809pOwl000 service=monitor error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:50.048718Z lvl=info msg="Reading file" log_id=0809pOwl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:14:50.363324Z lvl=info msg="Signal received, initializing clean shutdown..." log_id=0809pOwl000
ts=2018-05-11T19:14:50.363473Z lvl=info msg="Waiting for clean shutdown..." log_id=0809pOwl000
ts=2018-05-11T19:14:50.363636Z lvl=info msg="Listener closed" log_id=0809pOwl000 service=snapshot
ts=2018-05-11T19:14:50.425885Z lvl=info msg="Write failed" log_id=0809pOwl000 service=write shard=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:50.426024Z lvl=info msg="failed to store statistics" log_id=0809pOwl000 service=monitor error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:14:50.426090Z lvl=info msg="Shutting down monitor service" log_id=0809pOwl000 service=monitor
ts=2018-05-11T19:14:50.426163Z lvl=info msg="Terminating storage of statistics" log_id=0809pOwl000 service=monitor
ts=2018-05-11T19:14:50.426208Z lvl=info msg="Terminating precreation service" log_id=0809pOwl000 service=shard-precreation
ts=2018-05-11T19:14:50.426254Z lvl=info msg="Terminating continuous query service" log_id=0809pOwl000 service=continuous_querier
ts=2018-05-11T19:14:50.426303Z lvl=info msg="Closing retention policy enforcement service" log_id=0809pOwl000 service=retention
ts=2018-05-11T19:14:50.432726Z lvl=info msg="Closed service" log_id=0809pOwl000 service=subscriber
ts=2018-05-11T19:14:50.433336Z lvl=info msg="Server shutdown completed" log_id=0809pOwl000

ts=2018-05-11T19:15:00.242190Z lvl=info msg="InfluxDB starting" log_id=0809tS4W000 version=1.5.2 branch=1.5 commit=02d7d4f043b34ecb4e9b2dbec298c6f9450c2a32
ts=2018-05-11T19:15:00.242498Z lvl=info msg="Go runtime" log_id=0809tS4W000 version=go1.9.2 maxprocs=2
ts=2018-05-11T19:15:00.345745Z lvl=info msg="Using data dir" log_id=0809tS4W000 service=store path=/var/lib/influxdb/data
ts=2018-05-11T19:15:00.351405Z lvl=info msg="Open store (start)" log_id=0809tS4W000 service=store trace_id=0809tSVl000 op_name=tsdb_open op_event=start
ts=2018-05-11T19:15:00.469529Z lvl=info msg="Reading file" log_id=0809tS4W000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:15:00.652798Z lvl=info msg="Reading file" log_id=0809tS4W000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/telegraf/autogen/2/_00001.wal size=2391011
ts=2018-05-11T19:15:00.974440Z lvl=info msg="Failed to open shard" log_id=0809tS4W000 service=store trace_id=0809tSVl000 op_name=tsdb_open db_shard_id=2 error="[shard 2] sync /var/lib/influxdb/data/telegraf/autogen/2: invalid argument"
ts=2018-05-11T19:15:01.044153Z lvl=info msg="Failed to open shard" log_id=0809tS4W000 service=store trace_id=0809tSVl000 op_name=tsdb_open db_shard_id=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:15:01.044297Z lvl=info msg="Open store (end)" log_id=0809tS4W000 service=store trace_id=0809tSVl000 op_name=tsdb_open op_event=end op_elapsed=692.898ms
ts=2018-05-11T19:15:01.044466Z lvl=info msg="Opened service" log_id=0809tS4W000 service=subscriber
ts=2018-05-11T19:15:01.044566Z lvl=info msg="Starting monitor service" log_id=0809tS4W000 service=monitor
ts=2018-05-11T19:15:01.044739Z lvl=info msg="Registered diagnostics client" log_id=0809tS4W000 service=monitor name=build
ts=2018-05-11T19:15:01.044886Z lvl=info msg="Registered diagnostics client" log_id=0809tS4W000 service=monitor name=runtime
ts=2018-05-11T19:15:01.045003Z lvl=info msg="Registered diagnostics client" log_id=0809tS4W000 service=monitor name=network
ts=2018-05-11T19:15:01.045124Z lvl=info msg="Registered diagnostics client" log_id=0809tS4W000 service=monitor name=system
ts=2018-05-11T19:15:01.045358Z lvl=info msg="Starting precreation service" log_id=0809tS4W000 service=shard-precreation check_interval=10m advance_period=30m
ts=2018-05-11T19:15:01.045625Z lvl=info msg="Starting snapshot service" log_id=0809tS4W000 service=snapshot
ts=2018-05-11T19:15:01.045794Z lvl=info msg="Starting continuous query service" log_id=0809tS4W000 service=continuous_querier
ts=2018-05-11T19:15:01.045947Z lvl=info msg="Starting HTTP service" log_id=0809tS4W000 service=httpd authentication=false
ts=2018-05-11T19:15:01.046141Z lvl=info msg="opened HTTP access log" log_id=0809tS4W000 service=httpd path=stderr
ts=2018-05-11T19:15:01.045527Z lvl=info msg="Storing statistics" log_id=0809tS4W000 service=monitor db_instance=_internal db_rp=monitor interval=10s
ts=2018-05-11T19:15:01.046711Z lvl=info msg="Listening on HTTP" log_id=0809tS4W000 service=httpd addr=[::]:8086 https=false
ts=2018-05-11T19:15:01.046756Z lvl=info msg="Starting retention policy enforcement service" log_id=0809tS4W000 service=retention check_interval=30m
ts=2018-05-11T19:15:01.049235Z lvl=info msg="Sending usage statistics to usage.influxdata.com" log_id=0809tS4W000
ts=2018-05-11T19:15:01.049334Z lvl=info msg="Listening for signals" log_id=0809tS4W000
ts=2018-05-11T19:15:20.021776Z lvl=info msg="Reading file" log_id=0809tS4W000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=4228820
ts=2018-05-11T19:15:20.258580Z lvl=info msg="Write failed" log_id=0809tS4W000 service=write shard=1 error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"
ts=2018-05-11T19:15:20.259115Z lvl=info msg="failed to store statistics" log_id=0809tS4W000 service=monitor error="[shard 1] sync /var/lib/influxdb/data/_internal/monitor/1: invalid argument"

*Please note, the quickest way to fix a bug is to open a Pull Request.*

This appears related to the issue raised in: https://github.com/influxdata/influxdb/issues/9630

It appears to work okay if I use a volume for influxdb's data, but I am not quite sure where that data really gets stored and how reliable/easy to manage (size/backup) that will be.

### PR DESCRIPTION
**Pull request by zeebo**
multiple users have attempted to run influxdb in a docker container
with a windows host and a volume mounted from windows. that causes
problems because it apparently uses samba/cifs which does not
support fsync on directories. this patchset will, if it receives an EINVAL
on directory fsync, as is what appears to happen on samba/cifs, then it
 will ignore it. this should help.

fixes #9833.
fixes #9630.

### CODE REVIEW

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
``````

**Comment by zeebo** [comment]
caveat: i do not have a windows machine available, and so i have no idea if this will actually solve the problem. i at least tested that it continues to fsync on os x. hopefully we can have some more testing done.

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/178668772/src/github.com/influxdata/influxdb/cmd/influx-tools/importer/importer.go:73:3: this value of rpi is never used (SA4006)
```

File: pkg/file/file_unix.go [hunk 1]
```diff
@@ -11,7 +14,19 @@ func SyncDir(dirName string) error {
 		return err
 	}
 	defer dir.Close()
-	return dir.Sync()
+
+	// while we're on unix, we may be running in a docker container that is
```
Comment (jacobmarble) [inline pkg/file/file_unix.go]: Capitalize sentences

**Pull request merged**